source/common/common/utility.cc

```diff
@@ -214,4 +214,63 @@ std::string StringUtil::toUpper(const std::string& s) {
   return upper_s;
 }
 
+absl::string_view StringViewUtil::ltrim(absl::string_view source) {
+  source.remove_prefix(std::min(source.find_first_not_of(" \t\f\v\n\r"), source.size()));
+  return source;
+}
+
+absl::string_view StringViewUtil::rtrim(absl::string_view source) {
+  source.remove_suffix(source.size() - source.find_last_not_of(" \t\f\v\n\r") - 1);
+  return source;
+}
+
+absl::string_view StringViewUtil::trim(absl::string_view source) { return ltrim(rtrim(source)); }
+
+bool StringViewUtil::findToken(absl::string_view source, absl::string_view delimiters,
+                               absl::string_view key_token, bool trim_whitespace) {
+  const std::vector<absl::string_view> tokens = splitToken(source, delimiters, false);
```

Comment:
Your splitToken method already trims whitespace. Why don't you just pass trim_whitespace arg in there rather than trimming again in this method?

Reply:
I've already modified this in a upcoming PR for utility library :)
If trim_whitespace gets passed into splitToken(source, delimiters, false), all elements in the container gets trimmed. By doing that we might run into some unnecessary computation since what we need is to trim till a matched string and then return.